deps/src/main/java/com/microsoft/azure/sdk/iot/deps/transport/amqp/AmqpReactor.java

```diff
@@ -32,7 +32,11 @@ public void run() throws HandlerException
         this.reactor.setTimeout(REACTOR_TIMEOUT);
         this.reactor.start();
 
-        while(this.reactor.process()){}
+        //noinspection StatementWithEmptyBody
+        while(this.reactor.process())
+        {
+            // The empty while loop is to ensure that reactor thread runs as long as it has messages to process
```

Comment:
I didn't quite understand, what is the resolution of warning here?

Reply:
the warning has been suppressed by using the following statement:
`//noinspection StatementWithEmptyBody`

Reply:
that is because we do want an empty while block here, to simply wait until all processing has been completed (adding some kind of log statement to circumvent the warning wouldn't be the best solution here, since we'll end up with a lot of log statements and also cause cpu overhead)

Reply:
Agreed.